src/core_plugins/kibana/common/field_formats/types/_numeral.js

```diff
@@ -26,6 +26,15 @@ export class Numeral extends FieldFormat {
       }
 
       getParamDefaults() {
+        const language = this.getConfig('format:language');
```

Comment:
We really shouldn't be setting the `numeral` language here, this method is only meant to return the param defaults. This situation appears to be very similar to setting the timezone for `moment` which is being done in the `KibanaRootController` (here)[https://github.com/elastic/kibana/blob/master/src/core_plugins/kibana/public/kibana_root_controller.js#L9] but then we're also having to do so in other app controllers, [example here](https://github.com/elastic/kibana/blob/master/src/core_plugins/timelion/public/app.js#L71).

Perhaps @spalger can comment on a better place to put logic like this without these deficiencies.

Additionally, I have hesitations to setting the language for the global instance of numeral as this could have unintended side-effects outside of modifying the behavior of field formatters specifically.

Reply:
I agree it's probably not the best place to set this code. 
If you find a better place, I would be glad to hear it.
I set it there because this method is called at app loading beginning and is rarely called then.

Concerning your hesitations, currently, numeraljs allows only to set this setting globally, not locally for each formatter. Then, would it make sense to have several language formats on the same kibana app ?

Reply:
This thread is kind of merging with https://github.com/elastic/kibana/pull/14019#discussion_r139841833

Reply:
@spalger do you see a better place (method) to put the javascript code which changes numeral language ?

Reply:
I'd just put it in the _convert function, then adding per-field params is just about exposing the UI to do so.

Reply:
If I set it there, it would require static `import` of all languages.